### PR TITLE
Kubernetes Pod settings

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes

This PR adds `KubernetesPodSettings` and applies them to all Kubernetes-based orchestrators.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/kubernetes/pod_settings.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    Args:
+        dict_: Dict to serialize.
+
+    Returns:
+        The serialized dict.
+    """
+    return {key: _serialize(value) for key, value in dict_.items()}
+
+
+def _serialize(value: Any) -> Any:
```
Comment (htahir1) [inline src/zenml/integrations/kubernetes/pod_settings.py:121]: This simply seems liek too generic a function to be in `pod_settings.py`. Maybe refactor it out into utils, or rename to `_serialize_kubenertes`

**Review by htahir1 — APPROVED** [review]
Lots of things going on here but a quick looks yielded just one small nit which you cna ignore..

File: src/zenml/integrations/kubernetes/pod_settings.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    """Serializes any object.
```
Comment (fa9r) [inline src/zenml/integrations/kubernetes/pod_settings.py:122]: well not really *any* object ;)

Comment (schustmi) [reply src/zenml/integrations/kubernetes/pod_settings.py]: Updated to "any valid object" ;)

**Pull request merged**